runtime/src/accounts_db.rs

```diff
@@ -3078,30 +3082,32 @@ impl AccountsDB {
         if force_flush && num_slots_remaining >= FLUSH_CACHE_RANDOM_THRESHOLD {
             // Don't flush slots that are known to be unrooted
             let mut frozen_slots = self.accounts_cache.find_older_frozen_slots(0);
-            frozen_slots.retain(|s| *s > max_flushed_root);
+            frozen_slots.retain(|s| *s > max_previously_flushed_root);
             // Remove a random index 0 <= i < `frozen_slots.len()`
             let rand_slot = frozen_slots.choose(&mut thread_rng());
             if let Some(rand_slot) = rand_slot {
                 info!(
                     "Flushing random slot: {}, num_remaining: {}",
                     *rand_slot, num_slots_remaining
                 );
-                self.flush_slot_cache(*rand_slot, None::<&mut fn(&_, &_) -> bool>);
+                self.flush_slot_cache(*rand_slot, None::<&mut fn(&_, &_, &_) -> bool>);
             }
         }
     }
 
     fn flush_rooted_accounts_cache(
         &self,
-        mut max_flush_root: Option<Slot>,
+        max_flush_root: Option<Slot>,
         should_clean: Option<(&mut usize, &mut usize)>,
     ) -> (usize, usize) {
+        let mut max_clean_root = None;
         if should_clean.is_some() {
-            max_flush_root = self.max_clean_root(max_flush_root);
+            // If there is a long running scan going on, this could prevent any cleaning
+            // based on updates from slots > `max_clean_root`.
+            max_clean_root = self.max_clean_root(max_flush_root);
         }
```

Comment:
nits: 

```rust
let max_clean_root = should_clean.map(|_| {
    // If there is a long running scan going on, this could prevent any cleaning
    // based on updates from slots > `max_clean_root`.
    self.max_clean_root(max_flush_root);
});
```

(shorting code and reducing `mut` is important to reduce reader's cognitive load.)

Reply:
Also this code clearly reads/states that `max_clean_root` is defined at front and set in stone by narrowing `requested_flush_root` down via the `.max_clean_root()` filter _only if_ we `should_clean`.

Reply:
ah derp, yeah, thanks for cleaning this up